src/reanimated2/interpolation.js

```diff
@@ -1,30 +1,115 @@
+  'worklet';
+
+  return (
+    value === Extrapolate.EXTEND ||
+    value === Extrapolate.CLAMP ||
+    value === Extrapolate.IDENTITY
+  );
+}
+
+function validateType(type) {
+  'worklet';
+
+  if (!type && type !== undefined) {
+    throw new Error(
+      `Reanimated: config object is not valid please provide valid config, for example:
+       interpolate(value, [inputRange], [outputRange], {
+        extrapolateLeft: 'clamp',
+        extrapolateRight: 'extend',
+      }) or interpolate(value, [inputRange], [outputRange], 'clamp)`
+    );
+  }
+
+  if (
+    type?.constructor === Object &&
+    ((Object.keys(type).length === 2 &&
+      !('extrapolateLeft' in type && 'extrapolateRight' in type)) ||
+      (Object.keys(type).length === 1 &&
+        !('extrapolateLeft' in type || 'extrapolateRight' in type)) ||
+      Object.keys(type).length > 2)
+  ) {
+    throw new Error(
+      `Reanimated: config object is not valid please provide valid config, for example:
+       interpolate(value, [inputRange], [outputRange], {
+        extrapolateLeft: 'clamp',
+        extrapolateRight: 'extend',
+      })`
+    );
+  }
+
+  if (typeof type === 'object') {
+    if ('extrapolateLeft' in type && !isExtrapolate(type.extrapolateLeft)) {
+      throw new Error(
+        `Reanimated: not supported value for "extrapolateLeft" \nSupported values: ["extend", "clamp", "identity"]\n Valid example:
+         interpolate(value, [inputRange], [outputRange], {
+          extrapolateLeft: 'clamp',
+        })`
+      );
+    }
+
+    if ('extrapolateRight' in type && !isExtrapolate(type.extrapolateRight)) {
+      throw new Error(
+        `Reanimated: not supported value for "extrapolateRight" \nSupported values: ["extend", "clamp", "identity"]\n Valid example:
+         interpolate(value, [inputRange], [outputRange], {
+          extrapolateRight: 'clamp',
+        })`
+      );
+    }
+  }
+
+  if (typeof type === 'string' && !isExtrapolate(type)) {
+    throw new Error(
+      `Reanimated: not supported value for "interpolate" \nSupported values: ["extend", "clamp", "identity"]\n Valid example:
+       interpolate(value, [inputRange], [outputRange], "clamp")`
+    );
+  }
+}
+
+// TODO: support default values in worklets:
+// e.g. function interpolate(x, input, output, type = Extrapolate.CLAMP)
 function internalInterpolate(x, l, r, ll, rr, type) {
   'worklet';
   if (r - l === 0) return ll;
   const progress = (x - l) / (r - l);
   const val = ll + progress * (rr - ll);
   const coef = rr >= ll ? 1 : -1;
 
-  // TODO: support default values in worklets:
-  // e.g. function interplate(x, input, output, type = Extrapolate.CLAMP)
-  type = type || Extrapolate.EXTEND;
+  const config = { type, coef, val, ll, rr, x };
+
+  validateType(type);
+
+  if (typeof type === 'object') {
+    return coef * val < coef * ll
+      ? getVal({ ...config, type: type.extrapolateLeft })
+      : getVal({ ...config, type: type.extrapolateRight });
```

Comment:
I don't think that ternary replaces the original conditions, can you revert this? Or am I missing something?

Reply:
Done ✅